### PR TITLE
BracesFixer - add support for anonymous classes

### LINKED ISSUE #2030: Invalid anonymous class formating.
Anonymous class seems to loose indentation.

```
   1) unit/App/Remote/MessageSerializerTest.php
      ---------- begin diff ----------
      --- Original
      +++ New
      @@ @@
               $message = new class extends DomainMessage
      -        {
      -            
      -            public function __construct()
      -            {
      -                $this->init();
      -            }
      -            
      -            public function payload()
      -            {
      -                return [];
      -            }
      -            
      -            protected function setPayload(array $payload)
      -            {
      -                //
      -            }
      -            
      -            public function messageType()
      -            {
      -                return 'Test';
      -            }
      -        };
      + {
      +
      +     public function __construct()
      +     {
      +         $this->init();
      +     }
      +
      +     public function payload()
      +     {
      +         return [];
      +     }
      +
      +     protected function setPayload(array $payload)
      +     {
      +         //
      +     }
      +
      +     public function messageType()
      +     {
      +         return 'Test';
      +     }
      + };
       
               $SUT = new MessageSerializer();
               $SUT->messageToArray($message);
           }
       }
       
      
      ---------- end diff ----------
```

Config

```
<?php
$finder = Symfony\CS\Finder\DefaultFinder::create()
    ->in(__DIR__ . '/src')
    ->in(__DIR__ . '/tests');
$config = Symfony\CS\Config\Config::create();
$config->level(null);
$config->fixers(
    array(
        'braces',
        'duplicate_semicolon',
        'elseif',
        'empty_return',
        'encoding',
        'eof_ending',
        'function_call_space',
        'function_declaration',
        'indentation',
        'join_function',
        'line_after_namespace',
        'linefeed',
        'lowercase_constants',
        'lowercase_keywords',
        'parenthesis',
        'multiple_use',
        'method_argument_space',
        'object_operator',
        'php_closing_tag',
        'remove_lines_between_uses',
        'single_line_after_imports',
        'short_array_syntax',
        'short_tag',
        'standardize_not_equal',
        'trailing_spaces',
        'unused_use',
        'visibility',
        'whitespacy_lines',
    )
);
$config->finder($finder);
return $config;
```

Version of `php-cs-fixer`
```
vendor/bin/php-cs-fixer --version
PHP CS Fixer version 1.11.4 by Fabien Potencier
```

### PR DESCRIPTION
**Pull request by keradus**
Fixes #2030

### CODE REVIEW

File: Symfony/CS/Fixer/PSR2/BracesFixer.php [hunk 1]
```diff
@@ -429,6 +429,21 @@ private function detectIndent(Tokens $tokens, $index)
             if ($prevToken->isGivenKind($goBackTokens)) {
                 return $this->detectIndent($tokens, $prevIndex);
             }
+
+            if ($token->isClassy() && $prevToken->isGivenKind(T_NEW)) {
```
Comment (SpacePossum) [inline Symfony/CS/Fixer/PSR2/BracesFixer.php]: `isClassy` we only need T_CLASS check here I think?

File: Symfony/CS/Fixer/PSR2/BracesFixer.php [hunk 2]
```diff
@@ -429,6 +429,21 @@ private function detectIndent(Tokens $tokens, $index)
...
+                for ($prevIndex = $prevIndex - 1; 0 <= $prevIndex; --$prevIndex) {
+                    $prevToken = $tokens[$prevIndex];
+                    // if token is multiline whitespaces
+                    if ($prevToken->isWhitespace() && !$prevToken->isWhitespace(array('whitespaces' => " \t"))) {
```
Comment (SpacePossum) [inline Symfony/CS/Fixer/PSR2/BracesFixer.php]: why not test for `\n`, a serie of `\x0B` would pass the current check while it shouldn't

**Comment by SpacePossum** [comment]
thanks for picking this one up :)
two minor questions
:+1:

File: Symfony/CS/Fixer/PSR2/BracesFixer.php (hunk 2 above)
Comment (keradus) [inline Symfony/CS/Fixer/PSR2/BracesFixer.php]: seriously, do we support \0 as valid whitespace in the code ... ?

Comment (SpacePossum) [inline Symfony/CS/Fixer/PSR2/BracesFixer.php]: we do https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.11/Symfony/CS/Tokenizer/Token.php#L439
^_^

**Comment by SpacePossum** [comment]
:+1:  :ship:  :)

**Comment by keradus** [comment]
I have 2 options this morning: merge it as is or fix issues you have raised. Now, next window for merge is the evening ;)

**Comment by SpacePossum** [comment]
thats just perfect :)

**Pull request closed without merging**